frontend/src/app/AdminNavBar/AdminNavBar.tsx

```diff
@@ -78,13 +73,40 @@ export interface AdminNavBarProps {
 }
 
 export const AdminNavBar = ({ isMenuOpen }: AdminNavBarProps): JSX.Element => {
-  const { user } = useUser()
+  const { user, isLoading: isUserLoading } = useUser()
 
-  const {
-    isOpen: isContactModalOpen,
-    onClose: onContactModalClose,
-    onOpen: onContactModalOpen,
-  } = useDisclosure()
+  const emergencyContactKey = useMemo(
+    () => (user?._id ? EMERGENCY_CONTACT_KEY_PREFIX + user._id : null),
```

Comment:
We should also not show the modal if the user already has an emergency contact.

Reply:
ok updated! now emergencyContactKey is only saved in local storage if `user.id` exists and user has no contact